src/Map/MapPrimitiviser_P.cpp

```diff
@@ -2330,9 +2331,14 @@ void OsmAnd::MapPrimitiviser_P::obtainPrimitiveTexts(
         {
             const auto& caption = constOf(captions)[captionAttributeId];
             textEvaluator.setIntegerValue(env->styleBuiltinValueDefs->id_INPUT_TEXT_LENGTH, caption.length());
+            
+            QString tag = attributeMapping->decodeMap[captionAttributeId].tag;
+            if (tag == QString("name:en"))
+                nameEn = caption;
+            
             QString captionAttributeTag;
             if (captionAttributeId != attributeMapping->nativeNameAttributeId && captionAttributeId != localizedNameRuleId)
```

Comment:
Did you see https://github.com/osmandapp/OsmAnd-core/blob/d28f524fb2d34ff2880492d23f44ffe8b6b7f42d/src/Data/MapObject.cpp#L418 ?